documentation/proposals/Proposal - Region Specific Key Input.md

```diff
@@ -0,0 +1,105 @@
+# Summary
+- Make the windowing API return the key actually pressed, or any other key for that matter, instead of the key that is at
+the same location on a US-Keyboard.
+
+# Contributors
+- Matija Brown, None
+
+# Current Status
+- [x] Proposed
+- [ ] Discussed with API Review Board (ARB)
+- [ ] Approved
+- [ ] Implemented
+
+# Design Decissions
+- The main aspect is to change the original API as little as possible and also make the region-adapting
+toggleable / add the abillity to manually set a keyboard layout. To achieve this diversity,
+new layouts can be imported from an external file, but the included ones are hard coded. This way one can easilly switch between
+layouts and / or import a new, custome one while the rest of the input API remains unchanged. Due to the fact this is toggleable,
+it will also work the same way as before. The layout is saved in the IInputContext, as it is not possible to have different
+keyboard layouts on different keyboards at the same time, so it needn't be saved in every keyboard instance. If writing custom layouts
+turns out to be a common thing, there will be a layout generator created as well. For now key modifiers such as SHIFT will not be
+considered, but that might be implemented at a later time.
+
+# Proposed API
+
+## KeyboardLayout
+```cs
+public class KeyboardLayout
+{
+
+    /// <summary>
+    /// The different included layouts.
+    /// </summary>
+    public enum Layouts
+    {
+        /// <summary>
+        /// Automatically detect layout and use.
+        /// Works slightly differently to the hard-coded ones.
+        /// </summary>
+        Auto,
+
+        /// <summary>
+        /// The QWERTY layout
+        /// </summary>
+        QWERTY,
+        QÜERTY,
+        ÄWERTY,
+        QERTZ,
+        ...
+
+        NoLayout,
+        LastLayout
+    }
+
+    /// <summary>
+    /// The name of the Layout.
+    /// </summary>
+    public static string Name { get; }
```

Comment:
What does this do? is it just for displaying to the user?

Reply:
Yes, just for displaying to the user. I don't know why I made it static... sorry for the mistake, I'll change that.

Reply:
No, it's good 😄  Really appreciate your work!